scripts/build/termux_step_setup_variables.sh

```diff
@@ -37,6 +37,9 @@ termux_step_setup_variables() {
 
 	if [ "$TERMUX_PACKAGE_LIBRARY" = "glibc" ]; then
 		export TERMUX_PREFIX="$TERMUX_PREFIX/glibc"
+		if [ "$TERMUX_ON_DEVICE_BUILD" = "false" ] && [ "$TERMUX_APP_PACKAGE" != "com.termux" ]; then
```

Comment:
And here.

Reply:
To understand, this is where the glibc application environment prefix is ​​configured for cgct so that cgct can correctly build binaries. By default, cgct uses the termux prefix (`/data/data/com.termux/files/usr/glibc`) and essentially this algorithm compares `TERMUX_APP_PACKAGE` with the default part of cgct's prefix (let's assume it's called `CGCT_APP_PACKAGE`).

Reply:
Forkers should only ever have to modify the `properties.sh` file and not gave to look for and change hardcoded values in random files. If cgct requires a different prefix than normal termux, then:

- Add following to `properties.sh` and replace references of `CGCT_DIR` with `TERMUX_CGCT__TOOLS_DIR` (Use `ROOT_DIR` if that makes more sense for your reqs).
- Replace above with `[ "$TERMUX_APP_PACKAGE" != "$TERMUX_CGCT_APP__PACKAGE_NAME" ]`
- Replace `CGCT_APP_PREFIX` with `TERMUX_CGCT__PREFIX`.

```shell
# Path to CGCT tools.
TERMUX_CGCT_APP__PACKAGE_NAME="com.termux"
TERMUX_CGCT_APP__DATA_DIR="/data/data/$TERMUX_CGCT_APP__PACKAGE_NAME"
TERMUX_CGCT__TOOLS_SUBDIR="cgct"
TERMUX_CGCT__TOOLS_DIR="$TERMUX_CGCT_APP__DATA_DIR/$TERMUX_CGCT__TOOLS_SUBDIR"
export TERMUX_CGCT__TOOLS_DIR
```

Let me know if you want me to do it instead.

Reply:
In `clean.sh`, change shebang to `#!/bin/bash`, source `properties.sh`, and add replace `find /data -mindepth 1 ! -regex '^/data/data/com.termux/cgct\(/.*\)?' -delete` with following.

```shell
if [[ ! "$TERMUX_BASE_DIR" =~ ^(/[^/]+)+$ ]]; then
	echo "TERMUX_BASE_DIR '$TERMUX_BASE_DIR' is not an absolute path under rootfs '/'." 1>&2
	exit 1
fi
# Get `/path/` from `/path/to/rootfs`.
rootfs_top_level_dir="${TERMUX_BASE_DIR%"${TERMUX_BASE_DIR#/*/}"}"
find "$rootfs_top_level_dir" -mindepth 1 ! -path "$TERMUX_CGCT__TOOLS_DIR/*" -delete 2> /dev/null || true
```

Reply:
> Forkers should only ever have to modify the properties.sh file and not gave to look for and change hardcoded values in random files.

Forkers will not need to reconfigure these hardcoded values, since these values ​​represent the default setting in the cgct binaries.

> ```shell
> # Path to CGCT tools.
> TERMUX_CGCT_APP__PACKAGE_NAME="com.termux"
> TERMUX_CGCT_APP__DATA_DIR="/data/data/$TERMUX_CGCT_APP__PACKAGE_NAME"
> TERMUX_CGCT__TOOLS_SUBDIR="cgct"
> TERMUX_CGCT__TOOLS_DIR="$TERMUX_CGCT_APP__DATA_DIR/$TERMUX_CGCT__TOOLS_SUBDIR"
> export TERMUX_CGCT__TOOLS_DIR
> ```

I don’t see the point of specifying the path to the cgct instructions in such a complicated way. The only thing that probably makes sense is to create a variable `CGCT_DEFAULT_APP_PACKAGE` (the same as `TERMUX_CGCT_APP__PACKAGE_NAME`) which will represent the hardcoded value.

> Replace `CGCT_APP_PREFIX` with `TERMUX_CGCT__PREFIX`.

The meaning of the variable is a little lost.  If you use the `TERMUX_CGCT__PREFIX` variable, you might think that it is used somewhere in build scripts, although it is only used in cgct binaries. In addition, for this I will have to rebuild the cgct packages.

Reply:
Done